man/bootctl.xml

```diff
@@ -258,23 +258,60 @@
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>--entry-token=</option></term>
+
+        <listitem><para>Controls how to name and identify boot loader entries for this OS
+        installation. Accepted during <option>install</option>, and takes one of <literal>auto</literal>,
+        <literal>machine-id</literal>, <literal>os-id</literal>, <literal>os-image-id</literal> or an
+        arbitrary string prefixed by <literal>explicit:</literal> as argument.</para>
+
+        <para>If set to <option>machine-id</option> the entries are named after the machine ID of the running
+        systemd (e.g. <literal>b0e793a9baf14b5fa13ecbe84ff637ac</literal>). See
         <citerefentry><refentrytitle>machine-id</refentrytitle><manvolnum>5</manvolnum></citerefentry> for
         details about the machine ID concept and file.</para>
 
-        <para>Overriding this may be desirable to hide the machine ID from the (unencrypted) ESP, configure a
-        <citerefentry><refentrytitle>kernel-install</refentrytitle><manvolnum>8</manvolnum></citerefentry>
-        script, or, conversely, commit a transient machine ID.</para>
-
-        <para>The top-level machine ID directory is useful to allow smooth multi-boot installations: each
-        installed OS instance will have a different machine ID and thus a separate directory to place its
-        boot-time resources in. If this feature is turned off with this option, care needs to be taken that
-        multiple OS instances do not place conflicting files on the shared ESP and Extended Boot Loader
-        Partitions, or that multiple OS instances are not possible.</para></listitem>
+        <para>If set to <option>os-id</option> the entries are named after the OS ID of the running system,
+        i.e. the <varname>ID=</varname> field of
+        <citerefentry><refentrytitle>os-release</refentrytitle><manvolnum>5</manvolnum></citerefentry>
+        (e.g. <literal>fedora</literal>). Similar, if set to <option>os-image-id</option> the entries are
+        named after the OS image ID of the running system, i.e. the <varname>IMAGE_ID=</varname> field of
+        <filename>os-release</filename> (e.g. <literal>vendorx-cashier-system</literal>).</para>
+
+        <para>If set to <option>auto</option> (the default) it is first attempted to read the
+        <filename>/etc/kernel/entry-token</filename> file if it exists, and use the stored entry token
+        string. Otherwise if the local machine ID is initialized it is used. If it is not initialized the
+        <varname>IMAGE_ID=</varname> or <varname>ID=</varname> fields of <filename>os-release</filename> are
+        used, in this order of precedence, if set.</para>
+
+        <para>Unless set to <literal>machine-id</literal>, or when
+        <option>--make-entry-directory=yes</option> is used the selected token string is written
+        to a file <filename>/etc/kernel/entry-token</filename>, to ensure it will be used for future
```

Comment:
We had a report (https://github.com/systemd/systemd/issues/22224) complaining about /etc/kernel/entry-token being written even if kernel-install wasn't installed or used. So I figure we also shouldn't write /etc/kernel/entry-token when --make-entry-directory is set to os-id or os-image-id since kernel-install can find those directories on its own.

Reply:
You mean /etc/machine-info I presume? /etc/kernel/entry-token is a new thing this PR introduces after all.

It appears to me #22224 is not really something we support right now. We should, but that would involve adding a --root= switch, so that we operate on some other root fs.

Note that --make-entry-directory= defaults to no, so we don't actually create the dir at all right now, and hence also don't write the token out.

Reply:
Yeah that issue was complaining about /etc/machine-info being written, but I'd assume they'd have the same issue with /etc/kernel/entry-token being written when we land this PR.

Reply:
Anyway, I think this issue doesn't apply here, since we'll not create the dir by default anyway, it's opt-in.

Reply:
I had to spend a good 15 minute thinking about this before I figured out the intent behind it. I'd reword this sentence as follows:

> If not set to machine-id or when --make-entry-directory=yes is used, the selected token string ...